docs/spec.md

```diff
@@ -1292,20 +1294,21 @@ implementation-defined as well.
 
 #### Inputs
 
-| Label | Name      | Type   | Constraints |
-|-------|-----------|--------|-------------|
-| (I1)  | `operand` | tensor | (C1), (C2)  |
+| Label | Name      | Type                        | Constraints |
+|-------|-----------|-----------------------------|-------------|
+| (I1)  | `operand` | tensor or  quantized tensor | (C1-C2)     |
 
 #### Outputs
 
-| Name     | Type   | Constraints |
-|----------|--------|-------------|
-| `result` | tensor | (C1), (C2)  |
+| Name     | Type                        | Constraints |
+|----------|-----------------------------|-------------|
+| `result` | tensor or  quantized tensor | (C1-C2)     |
 
 #### Constraints
 
-* (C1) Given `E = element_type(operand)`, `E' = element_type(result)`, and
-  `R = rank(operand)`:
+* (C1) Given `E = is_quantized(operand) ? storage_type(operand) :
+  element_type(operand)`, `E' = is_quantized(result) ?
+  storage_type(result) : element_type(result)`, and `R = rank(operand)`:
   * If `num_bits(E') = num_bits(E)`, `shape(result) = shape(operand)`.
   * If `num_bits(E') < num_bits(E)`:
```

Comment:
Missed earlier, but there should be a constraint on remainder:

    `num_bits(E) % num_bits(E') == 0`

Reply:
Line 1314 as well, but in reverse:

    `num_bits(E') % num_ebits(E) == 0`

Reply:
Thanks @loganchien! That is a good point. I will resolve this in a separate [PR](https://github.com/openxla/stablehlo/pull/1637).

Reply:
Thanks @loganchien! That is a good point. I will resolve this in a separate [PR](https://github.com/openxla/stablehlo/pull/1637).